plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+import re
+import itertools
+import datetime
+import asyncio
+import logging
+import contextlib
+from functools import reduce
+from collections import namedtuple
+from enum import Enum, IntEnum
+from typing import List
+
+
+import discord
+
+import discord_client
+import util.discord
+from util.db.initialization import init_for
+from util import db
+
+import plugins.commands as commands
+import plugins.privileges as priv
+
+
+logger = logging.getLogger(__name__)
+
+# ---------- Constants ----------
+ticket_schema = """\
+CREATE SCHEMA tickets;
+
+
+CREATE TABLE tickets.tickets (
+        id 				SERIAL 	PRIMARY KEY,	-- Ticket id
+        type 			INT		NOT NULL,		-- Ticket type, referencing hard-coded enum
+        stage 			INT		NOT NULL,		-- Ticket stage, referencing hard-coded enum
```

Comment:
Looks like `stage` means in relation to being in a moderator's queue. I was honestly thinking this would be a separate table (for these queues). Is there a reason you want it to be in the same table?

Reply:
For simplicity of use? I see it as a property of the Ticket, and it's one column (two counting delivery id) to read/update, integrating into the ticket versioning system, rather than a new table.
I can move it into a separate table if it needs to be though.

Reply:
Not saying that it has to be. Could be like this if you want.